tools/pythonpkg/tests/fast/test_relation.py

```diff
@@ -153,6 +153,35 @@ def test_df_proj(self,duckdb_cursor):
         rel = duckdb.project(test_df, 'i')
         assert rel.execute().fetchall() == [(1,), (2,), (3,), (4,)]
 
+    def test_project_on_types(self, duckdb_cursor):
+        con = duckdb_cursor
+        con.sql("""
+            create table tbl(
+                c0 BIGINT,
+                c1 TINYINT,
+                c2 VARCHAR,
+                c3 TIMESTAMP,
+                c4 VARCHAR,
+                c5 STRUCT(a VARCHAR, b BIGINT)
+            )
+            """)
+        rel = con.table("tbl")
+        # select only the varchar columns
+        projection = rel.project(["varchar"])
+        assert projection.columns == ["c2", "c4"]
+
+        # select bigint, tinyint and a type that isn't there
+        projection = rel.project([duckdb.bigint, "tinyint", con.struct_type({'a': duckdb.varchar, 'b': duckdb.tinyint})])
```

Comment:
I think it would also be nicer if we could accept nested Python objects directly and map them, e.g.:

```py
{'a': str, 'b': int}
[int]
[[int]]
```

Reply:
I'd much prefer we use the Python notion here - `dict[str, str]`, `list[str]`, etc

Reply:
`dict[str, str]` would map to the `map` type, not to a struct (which requires fixed values for the keys, not just fixed types for the keys/values). `list[str]` looks great though.

Reply:
I'm not really sure what you mean by this, same as the other requests, wherever we accept a duckdb type, accept a nested python object and convert to the appropriate logical type where possible?

pybind11 should have support for implicit casting rules, same as we did for the ExplainType enum, shouldn't be too hard 👍

Reply:
> I'd much prefer we use the Python notion here - `dict[str, str]`, `list[str]`, etc

I'm not sure what you mean by this? accept these in string form and create duckdb types out of it?

Reply:
Still missing tests for these (I have them, but still need to write up proper pytests), but I added the logic 👍